### PR TITLE
Rework trampolines for POWER10

### PR DESCRIPTION
**Pull request by aviansie-ben**
This PR adds newly revamped trampolines when POWER10 support is enabled that use PC-relative prefixed load instructions rather than pTOC loads or address materialization sequences.

### CODE REVIEW

**Comment by gita-omr** [comment]
Jenkins test sanity aix,plinux jdk8,jdk11

File: runtime/compiler/runtime/Trampoline.cpp [hunk 1]
```diff
@@ -52,13 +55,12 @@ void ppcCodeCacheConfig(int32_t ccSizeInByte, int32_t *numTempTrampolines)
    {
    // Estimated: 2KB per method, with 10% being recompiled(multi-times)
```
[the remark this answers is not in the document]
Comment (zl-wang) [reply runtime/compiler/runtime/Trampoline.cpp:56]: no.  this is for temporary trampolines (as opposed to permanent trampolines). temporary trampoline is used for recompilation. at next GC point, temporary trampoline is collected and recycled, and the corresponding permanent trampoline is updated.

[the remark this answers is not in the document]
Comment (gita-omr) [reply runtime/compiler/runtime/Trampoline.cpp:56]: But why does it say 2KB while the shift right in the next line was changed from 12 to 13?

[the remark this answers is not in the document]
Comment (zl-wang) [reply runtime/compiler/runtime/Trampoline.cpp:56]: the original comment meant to say: on average, per compilation is 2KB. with 10% of them expected to be recompiled (and possibly multiple times), we are setting the number of temporary trampolines to be some multiple of ccSize/20KB. such that, the earlier version is ccSize/4KB (5 times recompilation).  This time around, the individual trampoline size is enlarged to 32byte (for better branch target prediction purpose). It is not affordable to have that many temporary trampolines.  5-time recompilation in a single GC period is too conservative and wasting temporary tramp[olines. As a result, we adjusted it down to 2.5 time recompilation ... so, ccSize/8KB.

[the remark this answers is not in the document]
Comment (zl-wang) [reply runtime/compiler/runtime/Trampoline.cpp:56]: by the way, running out of temporary trampoline is not a bug ... it still can continue:  just leading to repeated codePatching requests during that GC interval.  less performing in that sense.

[the remark this answers is not in the document]
Comment (zl-wang) [reply runtime/compiler/runtime/Trampoline.cpp:56]: i believed that (ccSize/8KB) is still plenty for temp trampolines.

File: runtime/compiler/runtime/Trampoline.cpp [hunk 2]
```diff
@@ -69,14 +71,54 @@ void ppcCreateHelperTrampolines(uint8_t *trampPtr, int32_t numHelpers)
...
+            buffer += 4;
+
+            // ori gr11, gr11, bits 48--63
+            *(int32_t *)buffer = 0x616b0000 | (helper & 0x0000ffff);
+            buffer += 4;
+            }
+         }
+      else
+         {
+         // pld gr11, [,16], 1 (PC-relative)
```
[the remark this answers is not in the document]
Comment (zl-wang) [reply runtime/compiler/runtime/Trampoline.cpp:109]: indeed, it is in each instruction comment.

**Pull request merged**